clarity/src/vm/clarity.rs

```diff
@@ -284,6 +293,7 @@ pub trait TransactionConnection: ClarityConnection {
     /// abort_call_back is called with an AssetMap and a ClarityDatabase reference,
     ///   if abort_call_back returns true, all modifications from this transaction will be rolled back.
     ///      otherwise, they will be committed (though they may later be rolled back if the block itself is rolled back).
+    #[allow(clippy::too_many_arguments)]
```

Comment:
I opened a refactoring issue for this function: https://github.com/stacks-network/stacks-core/issues/5943

I think it would have been nice if we did this _before_ this PR, because we could have avoided touching a bunch of files that don't really need to be updated, but I think it's a worthwhile refactor after this PR.

Reply:
I will work on 5943 soon after.